test/functional/lua/diagnostic_spec.lua

```diff
@@ -128,6 +128,26 @@ describe('vim.diagnostic', function()
     eq('Diagnostic #1', result[1].message)
   end)
 
+  it('clean the diagnostic cache when buf delete', function()
```

Comment:
This test still passes even without the changes in this PR. So it's not testing what you think it's testing.

Make sure the test fails on master, and passes when the changes from this PR are applied.

Reply:
done